### ISSUE TITLE
Should `next` require a space between its name and parentheses?

### ISSUE THREAD

**Issue by LukeGoodsell** [jimhester/lintr#294]
I think of `next` as a parameter-less function and so write things like

```
for (...) {
  if (...)
    next()
  ...
}
```

... but lintr is complaining that 

```
  XXX.R:160:13: style: Place a space before left parenthesis, except in a function call.
          next()
              ^
```

Is this intentional or a bug? I see from the code that `next` is not considered a function because it is neither a `SYMBOL_FUNCTION_CALL` nor a `FUNCTION`; it is a "`special`".

**Comment by jimhester** [comment]
Yes, and actually the code you are writing is not doing what you think. See the difference between the parse trees with a normal function call and `next()`. The latter is actually equivalent running the next statement in and constructing a call with the result.

```r
parse(text = "foo()")[[1]]
#> foo()
parse(text = "next()")[[1]]
#> (next)()
```

What you are writing works only because of the semantics of`next`, which skips over the remaining code in the block, it actually makes no sense in the language grammar.

I would recommend you remove the parentheses from `next`, which is a statement, not a call.

**Comment by LukeGoodsell** [comment]
Thanks, Jim, that's very interesting.
